File: getting-started/variant-go/module-05-functions-and-decomposition/exercise-03-file-organization/solution/format.go
package main

import (
	"fmt"
	"strings"
)

func formatCurrency(cents int) string {
	dollars := cents / 100
	remainder := cents % 100
	return fmt.Sprintf("$%d.%02d", dollars, remainder)
}

func formatStudentSummary(s Student) string {
	avg := averageScore(s.Scores)
	return fmt.Sprintf("%-10s %-25s avg=%.1f tuition=%s",
		s.Name, s.Email, avg, formatCurrency(s.Tuition))
}

func formatValidationErrors(name string, errs []string) string {
	return fmt.Sprintf("INVALID %-10s: %s", name, strings.Join(errs, "; "))
}

File: getting-started/variant-ts/module-05-functions-and-decomposition/exercise-03-file-organization/solution/format.js
import { averageScore } from "./student.js";

const formatCurrency = (cents) => {
  const dollars = Math.floor(cents / 100);
  const remainder = cents % 100;
  return `$${dollars}.${String(remainder).padStart(2, "0")}`;
};

export const formatStudentSummary = (student) => {
  const avg = averageScore(student.scores);
  return `${student.name.padEnd(10)} ${student.email.padEnd(25)} avg=${avg.toFixed(1)} tuition=${formatCurrency(student.tuition)}`;
};

export const formatValidationErrors = (name, errors) => {
  return `INVALID ${name.padEnd(10)}: ${errors.join("; ")}`;
};
